.github/workflows/lint.yml

```diff
@@ -10,7 +10,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        linter: [copyright, xmllint, cppcheck, cpplint]
+        linter: [copyright, cppcheck, cpplint, flake8, xmllint]
```

Comment:
separately did you want to add pre-commit config on this repo?

Reply:
Yeah these linter jobs are redundant, they're provided for quick feedback on linting status because otherwise you have to wait through the rosdep install and workspace build until the `colcon test` phase.

This repo already fully bought in to the ROS 2 core code style so probably not apply precommit our formatter configs - maybe change my mind on that later, i do like the Ruff & Clang Format combo a lot better, and precommit makes everything so smooth.

 But for starters the base set of precommit sanity cleaners could be nice